File: two_pointers/container_with_most_water.ts
function maxArea(height: number[]): number {
    let left = 0, right = height.length - 1;
    let maxWater = 0;

    while (left < right) {
        const width = right - left;
        const currentHeight = Math.min(height[left], height[right]);
        maxWater = Math.max(maxWater, width * currentHeight);

        if (height[left] < height[right]) left++;
        else right--;
    }

    return maxWater;
}

File: two_pointers/ContainerWithMostWater.java
public class ContainerWithMostWater {

    public static int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int maxWater = 0;

        while (left < right) {
            int width = right - left;
            int currentHeight = Math.min(height[left], height[right]);
            maxWater = Math.max(maxWater, width * currentHeight);

            if (height[left] < height[right])
                left++;
            else
                right--;
        }

        return maxWater;
    }
}
